pkgs/development/python-modules/swspotify/default.nix

```
@@ -0,0 +1,27 @@
+{ stdenv, lib, buildPythonPackage, fetchPypi, python3 }:
+
+buildPythonPackage rec {
+  pname = "SwSpotify";
+  version = "1.2.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0mrrg3zcm2ppxxzcva8nbyj1dkcplif4a88mvapphfd91a1d521q";
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [
+    requests flask-cors dbus-python
+  ];
+  
+  preConfigure = ''
+    substituteInPlace setup.py \
+      --replace 'requests>=2.24.0' 'requests~=2.23.0'
```

the difference is important, `~=2.23.0` is essentially `>=2.23,2.23.*` where as `~=2.23` is essentially `>=2.23,2.*`
```suggestion
      --replace 'requests>=2.24.0' 'requests~=2.23'
```